From 96effe3fd98e9dfe736d0477c0a60eb13fdf520a Mon Sep 17 00:00:00 2001
From: Christophe Kerello <christophe.kerello@foss.st.com>
Date: Tue, 6 Jun 2023 13:47:18 +0200
Subject: [PATCH 0333/1141] mmc: mmci: stm32: revert timeout calculation fix

This reverts commit 3388e55947c9e4ed230328f9155aed663db4722b.

Change-Id: Ia3a8414fd7f01a25c16658725934f7c4a8156948
Signed-off-by: Christophe Kerello <christophe.kerello@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/311069
Tested-by: Christophe KERELLO <christophe.kerello@st.com>
Reviewed-by: Christophe KERELLO <christophe.kerello@st.com>
Reviewed-by: Yann GAUTIER <yann.gautier@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
---
 drivers/mmc/host/mmci.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/drivers/mmc/host/mmci.c b/drivers/mmc/host/mmci.c
index 001f0926c29c..38a30ca56d87 100644
--- a/drivers/mmc/host/mmci.c
+++ b/drivers/mmc/host/mmci.c
@@ -1776,8 +1776,7 @@ static void mmci_set_max_busy_timeout(struct mmc_host *mmc)
 		return;
 
 	if (host->variant->busy_timeout && mmc->actual_clock)
-		max_busy_timeout = U32_MAX / DIV_ROUND_UP(mmc->actual_clock,
-							  MSEC_PER_SEC);
+		max_busy_timeout = ~0UL / (mmc->actual_clock / MSEC_PER_SEC);
 
 	mmc->max_busy_timeout = max_busy_timeout;
 }
-- 
2.39.2

